core/src/main/java/com/radixdlt/consensus/vertexstore/VertexStoreState.java

```diff
@@ -89,25 +89,18 @@
  * <p>In future, we'd like to move to having a separate vertex store, responsible for maintaining
  * its own state.
  */
-@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 @Immutable
 public final class VertexStoreState {
   private static final Logger logger = LogManager.getLogger();
 
   private final VertexWithHash root;
   private final HighQC highQC;
-  // TODO: collapse the following two
-  private final ImmutableList<VertexWithHash> vertices;
-  private final ImmutableMap<HashCode, VertexWithHash> idToVertex;
+  private final ImmutableSet<VertexWithHash> vertices;
```

Comment:
Perhaps we should adjust the `hashCode()` implementation of `VertexWithHash` as it's currently quite slow. Perhaps its hashcode and equal implementation should just use the `vertexHash`, if we assume it's trustworthy?

```java
  @Override
  public int hashCode() {
    return Objects.hash(this.vertexHash);
  }

  @Override
  public boolean equals(Object o) {
    if (o instanceof final VertexWithHash that) {
      return Objects.equals(this.vertexHash, that.vertexHash);
    }
    return false;
  }
```

Reply:
yup, sounds good :+1: I believe it is trustworthy